core/src/main/java/org/apache/calcite/rel/metadata/RelMdPredicates.java

```diff
@@ -966,15 +986,18 @@ protected EquivalenceFinder() {
      * b + b + e
      * </pre>
      */
-    class ExprsItr implements Iterator<Mapping> {
+    static class ExprsItr implements Iterator<Mapping> {
       final int[] columns;
       final BitSet[] columnSets;
       final int[] iterationIdx;
       @Nullable Mapping nextMapping;
       boolean firstCall;
+      int sourceCount;
+      int targetCount;
 
       @SuppressWarnings("JdkObsolete")
-      ExprsItr(ImmutableBitSet fields) {
+      ExprsItr(ImmutableBitSet fields, Map<Integer, BitSet> equivalence,
```

Comment:
is the supression of JdkObsolete still necessary?

Reply:
To be honest, I don't really understand the purpose of this annotation, so I just kept it as it is.

Reply:
```
@SuppressWarnings("JdkObsolete") is an annotation in Java that instructs the compiler to ignore warnings related to the usage of obsolete APIs from the JDK, such as old collection classes like Vector, Hashtable, or Enumeration. These APIs are considered outdated and are generally discouraged in favor of newer, more efficient alternatives like ArrayList, HashMap, or the Iterator interface.
```

Reply:
I don't see any such usage.
If you are using an IDE, you can remove the warning and see whether it complains.
Cleaning up code is always useful, even if it is not the goal of your PR.

Reply:
Thanks! I will try it.